packages/zoe/src/zoeService/instanceAdminStorage.js

```diff
@@ -1,51 +1,282 @@
+    },
+    makeUserSeat: (
+      { state, facets: { helper } },
+      invitationHandle,
+      initialAllocation,
+      proposal,
+      offerArgs = undefined,
+    ) => {
+      const { userSeat, zoeSeatAdmin } = makeZoeSeatAdminKit(
+        initialAllocation,
+        proposal,
+        helper,
+        state.zoeInstanceStorageManager.getWithdrawFacet(),
+      );
 
-  /** @type {GetInstallationForInstance} */
-  const getInstallationForInstance = async instance =>
-    instanceToInstanceAdmin.get(instance).getInstallationForInstance();
+      const seatHandle = makeSeatHandle();
+      state.seatHandleToZoeSeatAdmin.init(seatHandle, zoeSeatAdmin);
 
-  return harden({
-    getPublicFacet,
-    getBrands,
-    getIssuers,
-    getTerms,
-    getOfferFilter,
-    getInstallationForInstance,
-    getInstanceAdmin: instanceToInstanceAdmin.get,
-    initInstanceAdmin: instanceToInstanceAdmin.init,
-    deleteInstanceAdmin: instanceToInstanceAdmin.delete,
+      const seatData = harden({
+        proposal,
+        initialAllocation,
+        seatHandle,
+        offerArgs,
+      });
+
+      state.zoeSeatAdmins.add(zoeSeatAdmin);
+      assert(state.handleOfferObj, 'incomplete setup of zoe seat');
+      E.when(
+        E(state.handleOfferObj).handleOffer(invitationHandle, seatData),
+        ({ offerResultPromise, exitObj }) =>
+          zoeSeatAdmin.resolveExitAndResult(offerResultPromise, exitObj),
+        err => {
+          state.adminNode.terminateWithFailure(err);
+          throw err;
+        },
+      );
+
+      // return the userSeat before the offerHandler is called
+      return userSeat;
+    },
+    makeNoEscrowSeatKit: (
+      { state, facets: { helper } },
+      initialAllocation,
+      proposal,
+      exitObj,
+      seatHandle,
+    ) => {
+      const { userSeat, zoeSeatAdmin } = makeZoeSeatAdminKit(
+        initialAllocation,
+        proposal,
+        helper,
+        state.zoeInstanceStorageManager.getWithdrawFacet(),
+        exitObj,
+        true,
+      );
+
+      state.zoeSeatAdmins.add(zoeSeatAdmin);
+      state.seatHandleToZoeSeatAdmin.init(seatHandle, zoeSeatAdmin);
+      return { userSeat };
+    },
+    getOfferFilter: ({ state }) => state.offerFilterStrings,
+    setOfferFilter: ({ state }, strings) => {
+      assert(Array.isArray(strings), X`${q(strings)} must be an Array`);
+      const proposedStrings = harden([...strings]);
+      assert(
+        proposedStrings.every(s => typeof s === 'string'),
+        X`Blocked strings (${q(proposedStrings)}) must be an Array of strings.`,
+      );
+
+      state.offerFilterStrings = proposedStrings;
+    },
+    // if any string in the list matches, don't process the invitation. Strings
+    // that end in ':' may be prefix matches, others must match exactly.
+    isBlocked: ({ state }, string) => {
+      return state.offerFilterStrings.some(s => {
+        return string === s || (s.slice(-1) === ':' && string.startsWith(s));
+      });
+    },
```

Comment:
If I understand correctly, this function will return true when any item in `offerFilterStrings` either exactly matches the function input or ends with ":" and is a prefix of the function input (e.g., `isBlocked("foo:bar:baz")` will return true if `offerFilterStrings` contains either "foo:bar:baz" or "foo:" or "foo:bar:", but not if it contains only "bar" or "bar:").
```suggestion
    // If any offer filter string matches the input string, don't process the invitation.
    // Offer filter strings that end in ':' match if they are a prefix of the input string;
    // others must match it exactly.
    isBlocked: ({ state }, string) => {
      return state.offerFilterStrings.some(filterString => {
        return filterString === string || (filterString.endsWith(':') && string.startsWith(filterString));
      });
    },
```

Reply:
done.  Thanks.

There's no goal to support `"foo:bar:baz"`, but you're right that it works that way.